homeassistant/components/zha/light.py

```diff
@@ -250,6 +250,7 @@ async def async_turn_on(self, **kwargs: Any) -> None:
                 )
             )
             and brightness_supported(self._attr_supported_color_modes)
+            and not self._color_channel.execute_if_off_supported
```

Comment:
It shouldn't be needed to check if ``self._color_channel`` exists here, as this part can only be reached when ``xy_color``, ``hs_color``, or ``temperature`` is provided.

And it seems like that if the supported color modes don't include XY/HS/color_temp, HA strips unsupported parameters from a ``light.turn_on`` call.